_pyceres/core/crs_matrix.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "_pyceres/core/wrappers.h"
+#include "_pyceres/helpers.h"
+#include "_pyceres/logging.h"
+
+#include <Eigen/Sparse>
+#include <ceres/ceres.h>
+#include <ceres/crs_matrix.h>
+#include <pybind11/pybind11.h>
+
+namespace py = pybind11;
+
+namespace {
+py::tuple ConvertCRSToPyTuple(const ceres::CRSMatrix& crsMatrix) {
+  std::vector<int> rows, cols;
+  std::vector<double> values;
+
+  for (int row = 0; row < crsMatrix.num_rows; ++row) {
+    for (int k = crsMatrix.rows[row]; k < crsMatrix.rows[row + 1]; ++k) {
+      rows.push_back(row);
+      cols.push_back(crsMatrix.cols[k]);
+      values.push_back(crsMatrix.values[k]);
+    }
+  }
+
+  // return as a tuple
+  return py::make_tuple(rows, cols, values);
```

Comment:
Would it make sense to return numpy arrays instead? the conversion of C++ vectors to Python lists is pretty slow.

Reply:
Yes indeed. This is done.